incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -104,6 +105,12 @@ func (r *ObjectReconciler) SyncNamespace(ctx context.Context, log logr.Logger, n
 // update deletes this object if it's an obsolete copy, and otherwise ensures it's been propagated
 // to any child namespaces.
 func (r *ObjectReconciler) update(ctx context.Context, log logr.Logger, inst *unstructured.Unstructured) error {
+	//Checks to see if the object is correct and should propagate
+	if canPropagate := r.ensureCanPropagate(inst.GetNamespace(), inst); !canPropagate {
+		log.Info("Object cannot propagate")
+		return fmt.Errorf("Object has finalizers set")
```

Comment:
Returning an error means "the reconciler failed and needs to be retried." That's not what's happened here (the reconciler successfully determined that the object couldn't be propagated) so this should be `return nil`.

Reply:
I KNEW something was off when I had to learn a new way of sending an error!

I think I'm starting to understand code smell, now I just need to spot it more.